UOP1_Project/Assets/Scripts/AudioSystem.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using Random = UnityEngine.Random;
+
+public class AudioSystem : MonoBehaviour
+{
+
+	private bool _sfxMute;
+	private bool _musicMute;
+	private AudioSource _theme;
+	private AudioSource _sfx;
+	private static readonly List<string> mixBuffer = new List<string>();
+	private const float mixBufferClearDelay = 0.05f;
+
+	internal string currentTrack;
+
+	public float delayInCrossfading = 0.3f;
+	public List<Theme> tracks = new List<Theme>();
+	public List<SFX> sounds = new List<SFX>();
+
+	private static int BoolToBinary(bool b) => b ? 1 : 0;
+	private SFX GetSoundByName(string sName) => sounds.Find(x => x.name == sName);
+
+	public float MusicVolume => !PlayerPrefs.HasKey("MusicVolume") ? 1f : PlayerPrefs.GetFloat("MusicVolume");
+	public float SfxVolume => !PlayerPrefs.HasKey("SFXVolume") ? 1f : PlayerPrefs.GetFloat("SFXVolume");
```

Comment:
Better to move these strings into constants. There are few inside the `Awake` method too.